### DISCUSSION TITLE
There's no docs for all methods.

### DISCUSSION THREAD

**Discussion by cdiazr** [tectalichq/public-openai-client-php#32] [General] [answered]
In my case, I'm trying to create embeddings but I can't find de proper parameters.

**Comment by thejamescollins** [comment, marked as answer]
Hi there @cdiazr,

You're right - we do not have currently detailed documentation on all of OpenAI's endpoints and functionality.

Our suggestion is to look at [OpenAI's embeddings guide](https://platform.openai.com/docs/guides/embeddings) for overall details on how embedding creation works, and the [Embeddings API reference](https://platform.openai.com/docs/api-reference/embeddings).

You would then use the client's `embeddings->create()` method, along with a `Tectalic\OpenAi\Models\Embeddings\ CreateRequest ` model for the request. For example:

```php
/** @var \Tectalic\OpenAi\Models\Embeddings\CreateResponse $response */
$response = $openaiClient->embeddings()->create(
    new \Tectalic\OpenAi\Models\Embeddings\CreateRequest([
        ...
    ])
)->toModel();
```

Where `...` is an associative array of your Embeddings CreateRequest properties. 

Looking at https://github.com/tectalichq/public-openai-client-php/blob/main/src/Models/Embeddings/CreateRequest.php, the required properties are `model` and `input`.

So something like:

```php
/** @var \Tectalic\OpenAi\Models\Embeddings\CreateResponse $response */
$response = $openaiClient->completions()->create(
    new \Tectalic\OpenAi\Models\ Embeddings\CreateRequest([
        'model' => 'text-embedding-ada-002',
        'input' => 'Your input text',
    ])
)->toModel();
```

Hopefully that is enough to get you started.

All the best,

James
Tectalic
